### PR TITLE
separate CanBuildFrom for each SeqFactory

### PR DESCRIPTION
**Pull request by mkeskells**

### CODE REVIEW

**Comment by dwijnand** [comment]
What's the motivation behind this?  (just wondering)

**Comment by mkeskells** [comment]
> What's the motivation behind this? (just wondering)

we are building a custom implementation of an `IndexedSeq`, and we need to be able to know that the  return type required is an `IndexedSeq`, or a `Vector`. ATM you cant tell, as the `CanBuildFrom` is the same JVM instance

**Comment by lrytz** [comment]
LGTM, just `t6150.scala` needs a fix.

**Comment by mkeskells** [comment]
> LGTM, just `t6150.scala` needs a fix.

fixed and squashed

File: test/junit/scala/collection/immutable/ListTest.scala [hunk 1]
```diff
@@ -119,16 +119,21 @@ class ListTest extends AllocationTest {
...
     // - collection.Seq.ReusableCBF
+    // - collection.immutable.LinearSeq.ReusableCBF
+    // - collection.LinearSeq.ReusableCBF
     // are distinct all generate List instances ( and that other don't behave the same way)
     // If this changes then (at least) the method List#isLikeListReusableCBF should be changed accordingly
 
-    assertNotSame(List.ReusableCBF, collection.immutable.Seq.ReusableCBF)
-    assertNotSame(List.ReusableCBF, collection.Seq.ReusableCBF)
-    assertNotSame(collection.immutable.Seq.ReusableCBF, collection.Seq.ReusableCBF)
+    val allCompatibleCFBs = List[CanBuildFrom[_,_,_]](List.ReusableCBF,
```
Comment (hrhino) [inline test/junit/scala/collection/immutable/ListTest.scala]: CFB -> CBF

**Review by hrhino — APPROVED** [review]

File: test/junit/scala/collection/immutable/ListTest.scala (hunk 1 above)
Comment (mkeskells) [reply test/junit/scala/collection/immutable/ListTest.scala]: haha. Must teach Intellij to correct more of my misspellings

Comment (mkeskells) [reply test/junit/scala/collection/immutable/ListTest.scala]: fixed now

**Pull request merged**